amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/polling/SynchronousBlockingRetrievalFactory.java

```diff
@@ -47,14 +49,30 @@
     private final int maxRecords;
     private final Duration kinesisRequestTimeout;
 
-    public SynchronousBlockingRetrievalFactory(String streamName, KinesisAsyncClient kinesisClient, RecordsFetcherFactory recordsFetcherFactory, int maxRecords, Duration kinesisRequestTimeout) {
+    private Function<DataFetcherProviderConfig, DataFetcher> dataFetcherProvider;
+
+    public SynchronousBlockingRetrievalFactory(String streamName,
```

Comment:
Mark as deprecated?

Reply:
Would this be marked as deprecated? `PollingConfig.java` still uses the existing constructor to create a retrievalFactory when a dataFetcherProvider is not passed in